roles/configure_networking/tasks/configure_network.yml

```diff
@@ -37,6 +37,18 @@
     state: absent
   when:
     - ansible_os_family == "Debian"
+    - ( ansible_distribution_major_version | int ) <= 9
+
+- name: Adjusting /etc/network/interfaces
+  become: yes
+  ansible.builtin.lineinfile:
+    path: /etc/network/interfaces
+    regexp: ^source /etc/network/interfaces.d
+    line: source /etc/network/interfaces.d/*
+    state: present
+  when:
+    - ansible_distribution == "Debian"
```

Comment:
I am checking specifically for `ansible_distribution` and not `ansible_os_family` here to avoid any potential confusion with Ubuntu.